src/server/plugins/engine/components/EmailAddressField.ts

```diff
@@ -14,9 +15,25 @@ export class EmailAddressField extends TextField {
   constructor(def: EmailAddressFieldComponent, model: FormModel) {
     super(def, model)
 
-    const { schema, options } = def
+    const { schema, options, title } = def
 
-    this.formSchema = this.formSchema.email()
+    let formSchema = joi.string().label(title.toLowerCase()).email()
+
+    if (options.required === false) {
+      formSchema = formSchema.allow('').allow(null)
+    }
```

Comment:
Presume we should just tag onto `this.formSchema` from the extended text field?

All of these bits are set in `super(def, model)` although `.messages()` are different

```suggestion
```